pkg/api/dashboard.go

```diff
@@ -167,27 +169,28 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 	}
 
 	meta := dtos.DashboardMeta{
-		IsStarred:              isStarred,
-		Slug:                   dash.Slug,
-		Type:                   models.DashTypeDB,
-		CanStar:                c.IsSignedIn,
-		CanSave:                canSave,
-		CanEdit:                canEdit,
-		CanAdmin:               canAdmin,
-		CanDelete:              canDelete,
-		Created:                dash.Created,
-		Updated:                dash.Updated,
-		UpdatedBy:              updater,
-		CreatedBy:              creator,
-		Version:                dash.Version,
-		HasACL:                 dash.HasACL,
-		IsFolder:               dash.IsFolder,
-		FolderId:               dash.FolderId,
-		Url:                    dash.GetUrl(),
-		FolderTitle:            "General",
-		AnnotationsPermissions: annotationPermissions,
-		PublicDashboardEnabled: publicDashboardEnabled,
-		HasPublicDashboard:     hasPublicDashboard,
+		IsStarred:                           isStarred,
+		Slug:                                dash.Slug,
+		Type:                                models.DashTypeDB,
+		CanStar:                             c.IsSignedIn,
+		CanSave:                             canSave,
+		CanEdit:                             canEdit,
+		CanAdmin:                            canAdmin,
+		CanDelete:                           canDelete,
+		Created:                             dash.Created,
+		Updated:                             dash.Updated,
+		UpdatedBy:                           updater,
+		CreatedBy:                           creator,
+		Version:                             dash.Version,
+		HasACL:                              dash.HasACL,
+		IsFolder:                            dash.IsFolder,
+		FolderId:                            dash.FolderId,
+		Url:                                 dash.GetUrl(),
+		FolderTitle:                         "General",
+		AnnotationsPermissions:              annotationPermissions,
+		PublicDashboardEnabled:              publicDashboardEnabled,
+		HasPublicDashboard:                  hasPublicDashboard,
+		PublicDashboardTimeSelectionEnabled: publicDashboardTimeSelectionEnabled,
```

Comment:
The only option needed that I can see is "PublicDashboardEnabled"  (to know if we need to show the public badge in DashNav or not) 

HasPublicDashboard and  PublicDashboardTimeSelectionEnabled are not needed

Reply:
he api call that needs to set meta flags and public dashboard options is the api call to load a public dashboard 

https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/api/query.go#L31

and there the dashboard model can be modified as well (to hide time picker for example)